src/theory/quantifiers/sygus/cegis.cpp

```diff
@@ -235,10 +263,20 @@ bool Cegis::constructCandidates(const std::vector<Node>& enums,
     {
       std::vector<Node> fail_cvs = enum_values;
       Assert(candidates.size() == fail_cvs.size());
+      // try to solve entire problem?
       if (src->repairSolution(candidates, fail_cvs, candidate_values))
       {
         return true;
       }
+      Node rl = getRefinementLemmaFormula();
+      bool ret = false;
+      // try to solve for the refinement lemmas only
+      if (src->repairSolution(rl, candidates, fail_cvs, candidate_values))
+      {
+        // We will exclude the skeleton as well; this means that we have one
+        // chance to repair the solution.
+        ret = true;
```

Comment:
Why do you have this in an `if` block rather than `ret = src->repairSolution(rl, candidates, fail_cvs, candidate_values)`?

Reply:
I've refactored.